Lib/concurrent/futures/process.py

```diff
@@ -339,6 +346,8 @@ def shutdown_worker():
 
         # Release the queue's resources as soon as possible.
```

Comment:
Not a requirement for this PR, but it seems to me that the `_queue_management_worker` function has grown quite long and complicated. Do you think it could be turned into an object with a bunch of short and readable helper methods?

Reply:
Yes this would be a very much easier to read indeed. I will try to make a new PR with this change when I get a chance.

Reply:
Looking forward to your PR then @tomMoral :-)